pkg/apis/broker/v1beta1/broker_defaults.go

```diff
@@ -18,10 +18,40 @@ package v1beta1
 
 import (
 	"context"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	eventingduckv1beta1 "knative.dev/eventing/pkg/apis/duck/v1beta1"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+)
+
+var (
+	// DefaultBackoffDelay is the default backoff delay used in the backoff retry policy
+	// for the Broker delivery spec.
+	DefaultBackoffDelay = "PT1S"
+	// DefaultBackoffPolicy is the default backoff policy type used in the backoff retry
+	// policy for the Broker delivery spec.
+	DefaultBackoffPolicy = eventingduckv1beta1.BackoffPolicyExponential
+	// DefaultRetry is the default number of maximum delivery attempts for unacked messages
+	// before they are sent to a dead letter topic in the Broker delivery spec, in
+	// case a dead letter topic is specified. Without a dead letter topic specified,
+	// the retry count is infinite.
+	DefaultRetry int32 = 6
 )
 
 // SetDefaults sets the default field values for a Broker.
 func (b *Broker) SetDefaults(ctx context.Context) {
-	// The Google Cloud Broker doesn't have any custom defaults. The
-	// eventing webhook will add the usual defaults.
+	// Set the default delivery spec.
+	if b.Spec.Delivery == nil {
+		b.Spec.Delivery = &eventingduckv1beta1.DeliverySpec{}
+	}
+	if b.Spec.Delivery.BackoffPolicy == nil &&
+		b.Spec.Delivery.BackoffDelay == nil {
+		b.Spec.Delivery.BackoffPolicy = &DefaultBackoffPolicy
+		b.Spec.Delivery.BackoffDelay = &DefaultBackoffDelay
+	}
+	if b.Spec.Delivery.Retry == nil &&
+		(b.Spec.Delivery.DeadLetterSink != nil && !equality.Semantic.DeepEqual(b.Spec.Delivery.DeadLetterSink, &duckv1.Destination{})) {
```

Comment:
IIUC the second check `!equality.Semantic.DeepEqual(b.Spec.Delivery.DeadLetterSink, &duckv1.Destination{})` is not needed. Because the validation should reject empty sink.

Reply:
I think you're right, I'll fix that